zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/scaling/MarkPartitionBootstrappedProcessor.java

```diff
@@ -77,12 +93,29 @@ public void processDistributedCommand(final TypedRecord<ScaleRecord> command) {
     final var scalingKey = command.getKey();
     final var wasAlreadyBootstrapped = areAllPartitionsBootstrapped();
     stateWriter.appendFollowUpEvent(scalingKey, ScaleIntent.PARTITION_BOOTSTRAPPED, scaleUp);
+    // if the partition that has completed bootstrapping is the current one, resubscribe to all
+    // message start events and signals
+    if (scaleUp.getRedistributedPartitions().getFirst() == command.getPartitionId()) {
```

Comment:
:upside_down_face: Perhaps we could make this a bit more defensive and check for `contains` instead of using `getFirst`. I know that we currently only set a single partition id for this command but using `contains` just makes more sense for someone just reading this code.